src/vm/amd64/umthunkstub.S

```diff
@@ -0,0 +1,202 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Geoff Norton. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+.intel_syntax noprefix
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+//
+// METHODDESC_REGISTER: UMEntryThunk*
+//
+NESTED_ENTRY TheUMEntryPrestub, _TEXT, UMEntryPrestubUnwindFrameChainHandler
+    PUSH_ARGUMENT_REGISTERS
+    // +8 for alignment
+    alloc_stack (SIZEOF_MAX_FP_ARG_SPILL + 8)
+    SAVE_FLOAT_ARGUMENT_REGISTERS 0
+    END_PROLOGUE
+
+    mov rdi, METHODDESC_REGISTER 
+    call C_FUNC(TheUMEntryPrestubWorker)
+    
+    // we're going to tail call to the exec stub that we just setup 
+
+    RESTORE_FLOAT_ARGUMENT_REGISTERS 0
+    free_stack (SIZEOF_MAX_FP_ARG_SPILL + 8)
+    POP_ARGUMENT_REGISTERS
+    TAILJMP_RAX
+
+NESTED_END TheUMEntryPrestub, _TEXT 
+
+//
+// METHODDESC_REGISTER: UMEntryThunk*
+//
+NESTED_ENTRY UMThunkStub, _TEXT, UMThunkStubUnwindFrameChainHandler
+#define UMThunkStubAMD64_FIXED_STACK_ALLOC_SIZE (SIZEOF_MAX_INT_ARG_SPILL + SIZEOF_MAX_FP_ARG_SPILL + 24)
+#define UMThunkStubAMD64_XMM_SAVE_OFFSET 0x10
+#define UMThunkStubAMD64_INT_ARG_OFFSET 0x90
+
+// {optional stack args passed to callee}   <-- new RSP
+// [rare path spill area]                   <-- RBP
```

Comment:
[rare path spill area] and {padding for xmm alignment} should not be here. RBP should point to xmm0